src/modules/sway/bar.cpp

```diff
@@ -61,38 +66,108 @@ void BarIpcClient::onInitialConfig(const struct Ipc::ipc_response& res) {
 void BarIpcClient::onIpcEvent(const struct Ipc::ipc_response& res) {
   try {
     auto payload = parser_.parse(res.payload);
-    if (auto id = payload["id"]; id.isString() && id.asString() != bar_.bar_id) {
-      spdlog::trace("swaybar ipc: ignore event for {}", id.asString());
-      return;
-    }
-    if (payload.isMember("visible_by_modifier")) {
-      // visibility change for hidden bar
-      signal_visible_(payload["visible_by_modifier"].asBool());
-    } else {
-      // configuration update
-      auto config = parseConfig(payload);
-      signal_config_(std::move(config));
+    switch (res.type) {
+      case static_cast<uint32_t>(IPC_EVENT_WORKSPACE):
```

Comment:
Ouch. I made https://github.com/Alexays/Waybar/pull/1516 to fix the enum type.
Please, drop the casts once it's merged.

Reply:
Should have followed your suggestion literally :)

Reply:
Wasn't possible before the aforementioned PR; the compiler wouldn't allow you :)